Fix:

error: incompatible function pointer types assigning to 'xmlStructuredErrorFunc' (aka 'void (*)(void *, const _xmlError *)') from 'void (*)(void *, xmlErrorPtr)' (aka 'void (*)(void *, _xmlError *)')

https://github.com/zorba-processor/zorba/issues/22
https://github.com/zorba-processor/zorba/pull/23
--- src/store/naive/loader.h.orig	2015-06-16 06:26:38.000000000 -0500
+++ src/store/naive/loader.h	2024-11-14 20:54:06.000000000 -0600
@@ -111,7 +111,11 @@
       const zstring& docUri,
       std::istream& xmlStream) = 0;
 
-  static void error( void *ctx, xmlErrorPtr );
+#if LIBXML_VERSION >= 21200
+  static void error(void *ctx, const xmlError *error);
+#else
+  static void error(void *ctx, xmlError *error);
+#endif
 };
 
 
--- src/store/naive/loader_fast.cpp.orig	2015-06-16 06:26:38.000000000 -0500
+++ src/store/naive/loader_fast.cpp	2024-11-14 20:52:43.000000000 -0600
@@ -100,7 +100,11 @@
 /*******************************************************************************
 
 ********************************************************************************/
-void XmlLoader::error(void* ctx, xmlErrorPtr error)
+#if LIBXML_VERSION >= 21200
+void XmlLoader::error(void *ctx, const xmlError *error)
+#else
+void XmlLoader::error(void *ctx, xmlError *error)
+#endif
 {
   if ( error->level == XML_ERR_NONE )
     return;
